src/extra/components/xgm.py

```diff
@@ -353,7 +361,36 @@ def pulse_energy(self, sase=None):
 
             self._pulse_energy[pg] = pulse_energy
 
-        return self._pulse_energy[pg]
+        pulse_energy = self._pulse_energy[pg]
+        if series:
+            # Use .where() to convert 0's to nans, then convert to a Series,
+            # then drop the nan entries.
+            pulse_energy = pulse_energy.where(pulse_energy > 0).to_series().dropna()
+
+        return pulse_energy
+
+    def slow_train_energy(self, sase=None):
+        """Return the slow train energy from the XGM in microjoules.
+
+        This is an average pulse energy, averaged over all pulses for 10-20s.
+
+        Args:
+            sase (int): Same meaning as in
+                [XGM.pulse_energy()][extra.components.XGM.pulse_energy].
+        """
+        pg = self._check_sase_arg(sase)
+        if pg not in self._slow_train_energy:
+            if pg == PropertyGroup.MAIN:
+                key = "controlData.slowTrain"
+            else:
+                key = f"controlData.slowTrainSa{pg.value}"
+
+            energy = self.control_source[key].xarray()
```

Comment:
For scalar values, `pd.Series` may be preferred here as well, though of course it is less critical on a train boundary. It should be cheap to offer though.

Reply:
That's fine, no rush. We should discuss how we can try to make the types we return predictable in the long run. This componet is now out as it is anyway.